src/org/opendatakit/briefcase/ui/export/components/FormsTableViewModel.java

```diff
@@ -0,0 +1,163 @@
+package org.opendatakit.briefcase.ui.export.components;
+
+import static java.awt.Color.DARK_GRAY;
+import static java.awt.Color.GREEN;
+import static javax.swing.JOptionPane.getFrameForComponent;
+import static org.opendatakit.briefcase.ui.ScrollingStatusListDialog.showDialog;
+import static org.opendatakit.briefcase.ui.export.components.FormsTableView.EDITABLE_COLS;
+import static org.opendatakit.briefcase.ui.export.components.FormsTableView.HEADERS;
+import static org.opendatakit.briefcase.ui.export.components.FormsTableView.TYPES;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.swing.JButton;
+import javax.swing.table.AbstractTableModel;
+import org.opendatakit.briefcase.model.FormStatus;
+import org.opendatakit.briefcase.ui.export.ExportConfiguration;
+import org.opendatakit.briefcase.ui.export.ExportForms;
+
+class FormsTableViewModel extends AbstractTableModel {
+  private final List<Runnable> onChangeCallbacks = new ArrayList<>();
+  private final Map<FormStatus, JButton> detailButtons = new HashMap<>();
+  private final Map<FormStatus, JButton> confButtons = new HashMap<>();
+  private final ExportForms forms;
+
+  FormsTableViewModel(ExportForms forms) {
+    this.forms = forms;
+  }
+
+  public void onChange(Runnable callback) {
+    onChangeCallbacks.add(callback);
+  }
+
+  void refresh() {
+    confButtons.forEach((form, button) -> button.setEnabled(form.isSelected()));
+    detailButtons.forEach((form, button) -> button.setEnabled(!form.getStatusHistory().isEmpty()));
+    fireTableDataChanged();
+    triggerChange();
+  }
+
+  private void triggerChange() {
+    onChangeCallbacks.forEach(Runnable::run);
+  }
+
+  JButton buildDetailButton(FormStatus form) {
+    JButton button = new JButton("Details...");
+    button.setEnabled(false);
+    // Ugly hack to be able to use this factory in FormExportTable to compute its Dimension
+    if (form != null) {
+      button.addActionListener(__ -> {
+        button.setEnabled(false);
+        try {
+          showDialog(getFrameForComponent(button), form.getFormDefinition(), form.getStatusHistory());
+        } finally {
+          button.setEnabled(true);
+        }
+      });
+    }
+    return button;
+  }
+
+  JButton buildOverrideConfButton(FormStatus form) {
+    JButton button = new JButton("⚙");
```

Comment:
The ⚙ is not appearing for me.
![screen shot 2018-01-13 at 3 41 15 pm](https://user-images.githubusercontent.com/55099/34911141-6407003a-f878-11e7-8b4b-05edb1528378.png)

Reply:
It seems that Lucida Sans doesn't support `GEAR` unicode char... I will have to find an alternative...

Reply:
Any suggestions for buttons with icons in Swing? I'd prefer to use an icon for this to save some horizontal pixels on that column...

Reply:
Strangely, I can see the glyph on win10/java8...

Reply:
This seems to be a Mac issue. We can file separately.